Tests/MapboxMapsTests/Foundation/MapViewTests.swift

```diff
@@ -327,6 +331,23 @@ final class MapViewTests: XCTestCase {
         XCTAssertEqual(attributionURLOpener.openAttributionURLStub.invocations.count, 1)
         XCTAssertEqual(attributionURLOpener.openAttributionURLStub.invocations.first?.parameters, url)
     }
+
+    func testEventsFlushingOnDeinit() throws {
+        dependencyProvider.makeEventsManagerStub.returnValueQueue.append(EventsManagerStub())
+
+        autoreleasepool {
+            mapView = buildMapView()
+        }
+
+        let flushStub = try XCTUnwrap(mapView.eventsManager as? EventsManagerStub).flushStub
```

Comment:
I think it would be more logical to get a reference to the event manager mock at the beginning of the test

Reply:
It was my initial code 🙂 
But we do not need the eventsManager strong reference, as we do expect it to destroy along with MapView. 
But we do need a `flushStub` to get the number of calls to the flush method. Having a reference to EventsManager would stop it from deallocation so `flush` method would never call and test would fail.